app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,18 @@
+module SessionsHelper
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def log_out
+    session.delete :user_id
+    @current_user = nil
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id] if session[:user_id]
```

Comment:
Toán tử `||=` có tác dụng gì?

Reply:
`a ||= b` có nghĩa là:
- Nếu `a` chưa được khai báo hoặc `a` mang giá trị là `nil` hoặc `false` thì `a = b`
- Nếu `a` đã được khai báo hoặc `a` mang không mang giá trị `nil` hoặc `false` thì giữ nguyên giá trị của `a`
- Tương đương với: `a = a || b`
- Ở đây nếu đã tồn tại `@current_user` thì giữ nguyên. Nếu chưa tồn tại thì tìm trong database rồi gán vào `@current_user`